fuzz/src/erc20.rs

```diff
@@ -0,0 +1,276 @@
+#![cfg(feature = "solc-backend")]
+use fe_compiler_test_utils::*;
+use serde::{Deserialize, Serialize};
+// Create fuzz test for the contracts
+
+const ALICE: &str = DEFAULT_CALLER;
+const BOB: &str = "2000000000000000000000000000000000000002";
+const JAMES: &str = "3000000000000000000000000000000000000003";
+const TOTAL_SUPPLY: &str = "1000000000000000000000000";
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone, Copy)]
+pub struct TokenDetails<'a> {
+    pub name: &'a str,
+    pub symbol: &'a str,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone, Copy)]
+pub struct TransferInfo<'a> {
+    pub to: &'a str,
+    pub value: u128,
+}
+
+pub fn fe_erc20_constructor(details: TokenDetails) {
+    if !details.name.is_empty() {
+        with_executor(&|mut executor| {
+            let harness = deploy_contract(
+                &mut executor,
+                "test-utils/fuzz/fixtures/fe/erc20.fe", // TODO: Need to fix the path
+                "ERC20",
+                &[string_token(details.name), string_token(details.symbol)],
+            );
+            erc20_constructor(details, &mut executor, harness);
+        });
+    }
+}
+
+pub fn sol_erc20_constructor(details: TokenDetails) {
+    if !details.name.is_empty() {
+        with_executor(&|mut executor| {
+            let harness = deploy_contract(
+                &mut executor,
+                "fuzz/fixtures/solidity/erc20.sol",
+                "ERC20",
+                &[string_token(details.name), string_token(details.symbol)],
+            );
+            erc20_constructor(details, &mut executor, harness);
+        });
+    }
+}
+
+pub fn fe_erc20_transfer(info: TransferInfo) {
+    if info.value
+        <= TOTAL_SUPPLY
+            .parse::<u128>()
+            .unwrap_or(1000000000000000000000000)
+    {
+        with_executor(&|mut executor| {
+            let mut harness = deploy_contract(
+                &mut executor,
+                "test-utils/fuzz/fixtures/fe/erc20.fe", // TODO: Need to fix the path
+                "ERC20",
+                &[string_token("Fe Coin"), string_token("FE")],
+            );
+            erc20_transfer(info, executor, harness);
+        });
+    }
+}
+
+pub fn sol_erc20_transfer(info: TransferInfo) {
+    if info.value
+        <= TOTAL_SUPPLY
+            .parse::<u128>()
+            .unwrap_or(1000000000000000000000000)
```

Comment:
We should just `expect` here or make `TOTAL_SUPPLY` a `u128`.

Reply:
Okay, will add that change in another PR.